sql-serializer/src/main/java/io/cdap/pipeline/sql/serializer/bigquery/BigQuerySQLSerializer.java

```diff
@@ -61,139 +86,206 @@ public String getSQL(StructuredQuery query, Table temporaryTable) {
     }
     StringBuilder b = new StringBuilder();
     b.append("CREATE TEMPORARY TABLE ");
-    appendTableSQL(b, temporaryTable, false);
+    appendTable(b, temporaryTable);
     b.append(" AS ");
-    appendQuerySQL(b, query, false);
+    appendQuery(b, query);
+    b.append(';');
     return b.toString();
   }
 
-  private StringBuilder appendQuerySQL(StringBuilder b, StructuredQuery query, boolean isSubquery) {
-    // Keep track of which columns were defined in the current nesting level since they may also be used in filter
-    HashSet<Column> definedColumns = new HashSet<>();
-    if (isSubquery) {
-      b.append('(');
-    } else if (query.hasAlias()) {
-      throw new IllegalArgumentException("Cannot have an alias on the main outer query");
-    }
+  /**
+   * Takes an existing {@link StringBuilder} and appends the string form of a {@link StructuredQuery} to it.
+   * The structured query takes the form of:
+   *
+   * SELECT [{@link Column}][, Column]* FROM [{@link Queryable}][ WHERE [{@link Filter}]]?
+   *
+   * Example return for a simple query selecting columns CustomerName and Age from a table Customers where Age = 15:
+   * SELECT CustomerName, Age FROM Customers WHERE (Age = 15)
+   *
+   * @param b The existing string builder
+   * @param query The given query
+   * @return The string builder with the query appended
+   */
+  private StringBuilder appendQuery(StringBuilder b, StructuredQuery query) {
     b.append("SELECT ");
     for (int i = 0; i < query.getColumns().size(); i++) {
       Column col = query.getColumns().get(i);
+      appendColumn(b, col);
       if (col.hasAlias()) {
```

Comment:
I think it's better to put this in the appendColumn method, so that everything related to the column is contained in that one method. Unless there's a situation where you would append the column name and not the alias?

Reply:
I believe this is the case if we have a statement like:

`SELECT Name AS n FROM customers WHERE Name = "Dave";`

In this case, Name is the column object, but it is not appended in the filter.